src/components/route-book/index.tsx

```diff
@@ -0,0 +1,175 @@
+import { useBook } from '@/fetchers/book';
+import { fromBaseUnit } from '@/old/utils/math/hiLo';
+import { Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import {
+  Position,
+  PositionId,
+  PositionState_PositionStateEnum,
+} from '@penumbra-zone/protobuf/penumbra/core/component/dex/v1/dex_pb';
+import { Amount } from '@penumbra-zone/protobuf/penumbra/core/num/v1/num_pb';
+import { getDisplayDenomExponent } from '@penumbra-zone/getters/metadata';
+import { innerToBech32Address } from '@/old/utils/math/bech32';
+import { uint8ArrayToBase64 } from '@/old/utils/math/base64';
+import { useAssets } from '@/shared/state/assets';
+import { round } from '@/shared/round';
+import { useComputePositionId } from '@/shared/useComputePositionId';
+
+interface Route {
+  lpId: string;
+  position: Position;
+  price: number;
+  amount: number;
+}
+
+interface RouteWithTotal extends Route {
+  total: number;
+}
+
+/**
+ * This function loops over the route data and combines the routes with the same price
+ * and counts the total from the route’s amount
+ */
+function getTotals(data: Route[], isBuySide: boolean, limit: number): RouteWithTotal[] {
+  const totals = new Map<number, RouteWithTotal>();
+
+  data.forEach((route: Route) => {
+    const entry = totals.get(route.price);
+    if (entry) {
+      totals.set(route.price, {
+        ...route,
+        total: entry.total + route.amount,
+        amount: Math.min(entry.amount, route.amount),
+      });
+    } else {
+      totals.set(route.price, {
+        ...route,
+        total: route.amount,
+      });
+    }
+  });
+
+  return isBuySide
+    ? Array.from(totals.values())
+        .slice(0, limit)
+        .toSorted((a, b) => b.price - a.price)
+    : Array.from(totals.values()).slice(-limit);
+}
+
+function getDisplayData({
+  data,
+  computePositionId,
+  asset1,
+  asset2,
+  isBuySide,
+  limit,
+}: {
+  data: Position[];
+  computePositionId: ((position: Position) => PositionId) | undefined;
+  asset1: Metadata | undefined;
+  asset2: Metadata | undefined;
+  isBuySide: boolean;
+  limit: number;
+}): RouteWithTotal[] {
+  if (!computePositionId || !asset1 || !asset2) {
+    return [];
+  }
+
+  const asset1Exponent = getDisplayDenomExponent(asset1);
+  const asset2Exponent = getDisplayDenomExponent(asset2);
+
+  const getValue = (property: Amount | undefined, exponent: number) =>
+    fromBaseUnit(BigInt(property?.lo ?? 0), BigInt(property?.hi ?? 0), exponent);
+
+  const routes = data
+    // .filter(position => position.state?.state.toLocaleString() === 'POSITION_STATE_ENUM_OPENED')
+    .filter(position => position.state?.state === PositionState_PositionStateEnum.OPENED)
+    .map(position => {
+      const direction = position.phi?.pair?.asset2?.equals(asset1.penumbraAssetId) ? -1 : 1;
```

Comment:
question: does `-1` and `1` have any inherent meaning within trade routing? If not, think it would be easier to make this a string label like `primary` and `secondary` or something.